oci/client/delete.go

```diff
@@ -19,17 +19,44 @@ package client
 import (
 	"context"
 	"fmt"
+	"github.com/fluxcd/pkg/oci/auth/gcp"
 	"github.com/google/go-containerregistry/pkg/crane"
 	"github.com/google/go-containerregistry/pkg/name"
 )
 
 // Delete deletes a particular image from an OCI repository
-// If the url has no tag, the latest image is deleted
-func (c *Client) Delete(ctx context.Context, url string) error {
-	_, err := name.ParseReference(url)
+// For GCR, GAR, the tag is deleted first, then the image
+// as it doesn't let you delete an image if a tag still references it.
+// This doesn't work with ECR/GHCR since they don't support DELETE according
+// to the docker API spec.
+func (c *Client) Delete(ctx context.Context, url string, tag string) error {
```

Comment:
You are right about removing the tag argument but I think it might best to not add deleting by using digest as an option, since we always delete using the digest (GCP doesn't permit deletion if the image still has a tag, and deleting the tag only deletes the tag, not the image itself. Azure doesn't allow deletion using tags).

We could still add a delete option but right now I am not sure what should be in it.
@stefanprodan suggested putting this issue on hold while we figure out if we do want to add this feature and how best to go about it.